src/vs/base/parts/quickopen/browser/quickOpenWidget.ts

```diff
@@ -410,14 +410,19 @@ export class QuickOpenWidget implements IModelProvider {
 
 		// Trigger open of element on selection
 		if (this.isVisible()) {
-			const context: IEntryRunContext = { event, keymods: this.extractKeyMods(event), quickNavigateConfiguration: this.quickNavigateConfiguration };
+			let eventForContext = event;
 			let mode = Mode.OPEN;
 
-			if (context.event instanceof StandardKeyboardEvent) {
-				if (context.event.keyCode === KeyCode.RightArrow) {
+			if (event instanceof StandardKeyboardEvent) {
+				eventForContext = event.browserEvent;
```

Comment:
@bpasero preserving the browser's event as the context here.  will this work better?  allows us to leave `Mode` logic in `elementSelected`.